cfpb/sbl-frontend
# [e2e] speed up `unavailableApis.spec.ts`, make api calls retries env variable

closes #916
closes #993

## Changes
- dev: Adds `SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS` as an environment variable
- dev: Disables `React-Query` _retries_ based on `SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS`
- e2e: Updates `unavailableApis.spec.ts` to not wait for retries
- e2e: further utilized `uploadFile.ts` to deduplicate code usage
- e2e: `yarn test:e2e` edited to include chmod and `./run-playwright.sh`

## How to test this PR
1. Add `SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS="false"` to your `.env`
2. ~~`chmod +x run-playwright.sh`~~
3. ~~`sh run-playwright.sh` or `sh run-playwright.sh --headless`~~
4. Run `yarn test:e2e`
5. Run `unavailableApis.spec.ts`
6. Verify the test passes and that the blocked APIs show up only once (see screenshot)

## Screenshots
<img width="1392" alt="Screenshot 2024-10-03 at 3 57 20 PM" src="https://github.com/user-attachments/assets/de80dbf5-458a-484b-89e8-8a8c68de9d92">


## Notes
- ~~The tester must now manually set `SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS` to `true` before running tests~~
- The script `run-playwright.sh`  is used so that the tester doesn't need to manually flip `SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS` before tests

## Questions
- Do we want an actual environment variable to manually control the amount of `React-Query` retries for all requests or is this approach (i.e. `SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS` -- a test flag) sufficient?
